### PR TITLE
Fix custom editor priority

### PR DESCRIPTION
**Pull request by akurinnoy**
### What does this PR do?

This PR fixes the priority of defining a custom editor. 
So the lowest priority has the `defaultEditor` parameter set in the CheCluster CR. The next option with higher priority is `.che/che-editor.yaml` file. The highest priority has the `che-editor` URL parameter of a factory link.

### What issues does this PR fix or reference?

fixes https://issues.redhat.com/browse/CRW-3921

### Is it tested? How?

1. Try `https://<che-host>#https://github.com/akurinnoy/vigilant-umbrella?che-editor=eclipse/che-theia/latest`. In this case Che-Theia editor will be used because the URL param has the highest priority.
2. Try `https://<che-host>#https://github.com/akurinnoy/vigilant-umbrella`. You'll get the Che-Idea because it's defined in the `.che/che-editor.yaml` file.
3. Try `https://<che-host>#https://github.com/eclipse-che/che-dashboard`. The default editor will be used.

### CODE REVIEW

**Review by ibuziuk — APPROVED** [review]
:+1:

File: packages/dashboard-frontend/src/store/Workspaces/devWorkspaces/index.ts [hunk 1]
```diff
@@ -31,16 +31,20 @@ import { WorkspaceAdapter } from '../../../services/workspace-adapter';
...
   IStatusUpdate,
 } from '../../../services/workspace-client/devworkspace/devWorkspaceClient';
 import { selectRunningWorkspacesLimit } from '../../ClusterConfig/selectors';
 import { createObject } from '../../helpers';
 import { selectDefaultNamespace } from '../../InfrastructureNamespaces/selectors';
-import * as DwPluginsStore from '../../Plugins/devWorkspacePlugins';
-import { selectDwEditorsPluginsList } from '../../Plugins/devWorkspacePlugins/selectors';
+import * as PluginsStore from '../../Plugins/devWorkspacePlugins';
+import {
+  selectDefaultEditor,
```
Comment (olexii4) [inline packages/dashboard-frontend/src/store/Workspaces/devWorkspaces/index.ts:42]: Do we really need both  'selectDefaultEditor' and 'selectDefaultEditorCR'?

Comment (akurinnoy) [reply packages/dashboard-frontend/src/store/Workspaces/devWorkspaces/index.ts:42]: You're right, now I see `selectDefaultEditorCR` is redundant.

**Review by ibuziuk — APPROVED** [review]
verified against https://github.com/dipakkarde/che-demo with overriding `che-editor=che-incubator/che-code/insiders` works like a charm :+1:

**Review by olexii4 — APPROVED** [review]

**Pull request merged**